Start Menu and Persistent Level controller

cjh15hub opened the pull request:
- Add Start Menu
- Add Start Scene
- Add Persistent Level Controller Logic

cjh15hub committed:
Level Controller

Add Additional Features to Level Controller
Add Start Menu
Add Start Scene

cjh15hub committed:
merge

cjh15hub committed:
merge

ryanc16 commented on Assets/Scripts/LevelController.cs:
```diff
@@ -15,20 +15,21 @@ public class LevelController : MonoBehaviour
     {
+        currentLevel = SceneManager.GetActiveScene();
         monsters = FindObjectsOfType<Monster>();
+        Debug.Log("monsters: " + monsters.Length);
```
Are we keeping `Debug.Log` entries in merged code or was this just for testing?

cjh15hub replied:
Removed

ryanc16 commented on Assets/Scripts/Menu/MenuBird.cs:
```diff
@@ -0,0 +1,32 @@
+
+    public void Update()
+    {
+        if(direction >= 1)
```
Could you change the `direction` to a type `Vector2` and then have the bird always translating using `direction * Time.deltaTime * speed`, then when the bird reaches the edges of the screen flip the direction between `Vector2.right` and `Vector2.left`.

cjh15hub replied:
They don't turn around, they pop onto the opposite side of the screen, resetting their path. But I can improve the code slightly by using a velocity vector instead and always translating on that.

ryanc16 commented on Assets/Scripts/Menu/MenuBird.cs:
```diff
@@ -0,0 +1,32 @@
+    {
+        if(direction >= 1)
+        {
+            transform.Translate(Vector3.right * Time.deltaTime * speed);
```
Shouldn't this be a `Vector2`?

cjh15hub committed:
Resolve PR Comments

- Remove Debug Log Statement
- Simplify Menu Bird translate logic
- Allow either Start Scene to load the other

ryanc16 commented on Assets/Scripts/Menu/MenuBird.cs:
```diff
@@ -0,0 +1,19 @@
+{
+    public Vector2 velocity = new Vector2(2.0f, 0);
+
+    private float screenEnds = 20;
```
Would it be possible to derive this number via the camera view somehow to make it more dynamic and tied to the actual viewport?

cjh15hub replied:
I have updated the script to derive the off screen positions of the camera

ryanc16 commented on Assets/Scripts/Menu/MenuBird.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public void Update()
+    {
+        if (transform.position.x > screenEnds) transform.position = new Vector2(-1 * screenEnds, transform.position.y);
```
Going off comment above, if there is someway to get the viewport width, a variable could be created for both the left and right viewport limits and calculated once ahead of time.
```cs
private float screenRight = camera.viewport.width/2;
private float screenLeft = -1 * screenRight;
.
.
.
if (transform.position.x > screenRight) ...
else if (transform.position.x < screenLeft) ...
```

ryanc16 commented on Assets/Scripts/Menu/MenuBird.cs:
```diff
@@ -0,0 +1,19 @@
+    public void Update()
+    {
+        if (transform.position.x > screenEnds) transform.position = new Vector2(-1 * screenEnds, transform.position.y);
+        if (transform.position.x < -1 * screenEnds) transform.position = new Vector2(screenEnds, transform.position.y);
```
This should probably be an `else if`. I can't see how it could ever be both in the same Update

cjh15hub committed:
Improve Camera Bird Script

Improve Camera Bird Script to calculate offScreen positions
